### PR TITLE
IPv4 Randomize SNAT support for IPv6 pods

### PR DESCRIPTION
**Pull request by achevuru**
**What type of PR is this?**
Enhancement + Addresses few inconsistencies/bugs with IPv6 mode.

**Which issue does this PR fix**:
NA

**What does this PR do / Why do we need it**:
- For Egress IPv4 flows from an IPv6 pod, we SNAT the node-local IPv4 address assigned to an IPv6 Pod to Node's Primary IPv4 address. We enable `random-fully` by default if the underlying iptables version supports it. PR extends support for `AWS_VPC_K8S_CNI_RANDOMIZESNAT` to the `egress-v4-cni-plugin` which will allow user to configure this.
- Addresses duplicate IPv4 route issue across CNI restarts in IPv6 mode.

**Testing done on this change**:
 - Validated that `random-fully` option is only set if `AWS_VPC_K8S_CNI_RANDOMIZESNAT` is set to `prng` for egress IPv4 flows.
 - Validated that the node local IPv4 route isn't replicated across CNI restarts.
 - 
**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
 - Extends `AWS_VPC_K8S_CNI_RANDOMIZESNAT` support to `egress-v4-cni-plugin`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/egress-v4-cni-plugin/snat/snat.go [hunk 1]
```diff
@@ -43,13 +43,17 @@ func iptRules4(target, src net.IP, chain, comment string, useRandomFully bool) [
...
-func Snat4(target, src net.IP, chain, comment string) error {
+func Snat4(target, src net.IP, chain, comment, randomizeSNAT string) error {
 	ipt, err := iptables.NewWithProtocol(iptables.ProtocolIPv4)
 	if err != nil {
 		return fmt.Errorf("failed to locate iptables: %v", err)
 	}
 
-	rules := iptRules4(target, src, chain, comment, ipt.HasRandomFully())
+	useRandomFully := true
+	if randomizeSNAT == "none" {
```
Comment (M00nF1sh) [inline cmd/egress-v4-cni-plugin/snat/snat.go:53]: 1. shall we support for "--random" as well? i.e. randomizeSNAT == "hashrandom"
2. we should check ipt.HasRandomFully() as well, and if it's not supported, fallback to either "--random" or none. (depends on question 1)

**Review by M00nF1sh — APPROVED** [review]
/lgtm

**Review by jayanthvn — APPROVED** [review]
lgtm, but I think we should add some documentation around this, I mean explaining how this works.

**Comment by achevuru** [comment]
> lgtm, but I think we should add some documentation around this, I mean explaining how this works.
Behavior currently documented under `AWS_VPC_K8S_CNI_RANDOMIZESNAT` will still be correct. We're extending the env variable out to the secondary CNI plugin as well. We can add a note if required under that. Will track it for README updates prior to 1.11 release.

**Comment by jayanthvn** [comment]
> > lgtm, but I think we should add some documentation around this, I mean explaining how this works.
> 
> 
> 
> Behavior currently documented under `AWS_VPC_K8S_CNI_RANDOMIZESNAT` will still be correct. We're extending the env variable out to the secondary CNI plugin as well. We can add a note if required under that. Will track it for README updates prior to 1.11 release.

Sounds good.

**Pull request merged**